etcdserver/server.go

```diff
@@ -1024,8 +1024,37 @@ func (s *EtcdServer) parseProposeCtxErr(err error, start time.Time) error {
 		if start.After(prevLeadLost) && start.Before(curLeadElected) {
 			return ErrTimeoutDueToLeaderFail
 		}
+
+		lead := types.ID(atomic.LoadUint64(&s.r.lead))
+		switch lead {
+		case types.ID(raft.None):
+			// TODO: return error to specify it happens because the cluster cannot elect leader
```

Comment:
if it has no leader, it means that it is candidate state. so in its view the cluster cannot elect a leader.